src/Microsoft.CodeQuality.Analyzers/Core/Security/MarkVerbHandlersWithValidateAntiforgeryTokenAnalyzer.cs

```diff
@@ -0,0 +1,360 @@
+
+        private static readonly LocalizableString GetAndTokenMessage = new LocalizableResourceString(
+            nameof(MicrosoftSecurityAnalyzersResources.MarkVerbHandlersWithValidateAntiforgeryTokenGetAndTokenMessage),
+            MicrosoftSecurityAnalyzersResources.ResourceManager,
+            typeof(MicrosoftSecurityAnalyzersResources));
+
+        private static readonly LocalizableString GetAndOtherAndTokenMessage = new LocalizableResourceString(
+            nameof(MicrosoftSecurityAnalyzersResources.MarkVerbHandlersWithValidateAntiforgeryTokenGetAndOtherAndTokenMessage),
+            MicrosoftSecurityAnalyzersResources.ResourceManager,
+            typeof(MicrosoftSecurityAnalyzersResources));
+
+        private static readonly LocalizableString VerbsAndNoTokenMessage = new LocalizableResourceString(
+            nameof(MicrosoftSecurityAnalyzersResources.MarkVerbHandlersWithValidateAntiforgeryTokenVerbsAndNoTokenMessage),
+            MicrosoftSecurityAnalyzersResources.ResourceManager,
+            typeof(MicrosoftSecurityAnalyzersResources));
+
+        private static readonly DiagnosticDescriptor NoVerbsRule = new DiagnosticDescriptor(
+            RuleId,
+            Title, 
+            NoVerbsMessage, 
+            DiagnosticCategory.Security,
+            DiagnosticSeverity.Warning, 
+            isEnabledByDefault: false);
+
+        private static readonly DiagnosticDescriptor NoVerbsNoTokenRule = new DiagnosticDescriptor(
+            RuleId,
+            Title,
+            NoVerbsNoTokenMessage,
+            DiagnosticCategory.Security,
+            DiagnosticSeverity.Warning,
+            isEnabledByDefault: false);
+
+        private static readonly DiagnosticDescriptor GetAndTokenRule = new DiagnosticDescriptor(
+            RuleId,
+            Title,
+            GetAndTokenMessage,
+            DiagnosticCategory.Security,
+            DiagnosticSeverity.Warning,
+            isEnabledByDefault: false);
+
+        private static readonly DiagnosticDescriptor GetAndOtherAndTokenRule = new DiagnosticDescriptor(
+            RuleId,
+            Title,
+            GetAndOtherAndTokenMessage,
+            DiagnosticCategory.Security,
+            DiagnosticSeverity.Warning,
+            isEnabledByDefault: false);
+
+        private static readonly DiagnosticDescriptor VerbsAndNoTokenRule = new DiagnosticDescriptor(
+            RuleId,
+            Title,
+            VerbsAndNoTokenMessage,
+            DiagnosticCategory.Security,
+            DiagnosticSeverity.Warning,
+            isEnabledByDefault: false);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(NoVerbsRule, NoVerbsNoTokenRule, GetAndTokenRule, GetAndOtherAndTokenRule, VerbsAndNoTokenRule);
+
+        /// <summary>
+        /// Helper for examining System.Web.Mvc attributes on MVC controller methods.
+        /// </summary>
+        private class MvcAttributeSymbols
```

Comment:
Nit: sealed?

Reply:
Also consider moving this nested type to a separate source file as it seems to be significantly large to deserve it's own file.